Difi.Oppslagstjeneste.Klient/OppslagstjenesteProxy.cs

```diff
@@ -0,0 +1,126 @@
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using Difi.Felles.Utility;
+using Difi.Oppslagstjeneste.Klient.Domene.Exceptions;
+using Difi.Oppslagstjeneste.Klient.Envelope;
+using Difi.Oppslagstjeneste.Klient.Svar;
+using Difi.Oppslagstjeneste.Klient.XmlValidering;
+
+namespace Difi.Oppslagstjeneste.Klient
+{
+    internal class OppslagstjenesteProxy
+    {
+        private readonly OppslagstjenesteInstillinger _oppslagstjenesteInstillinger;
+
+        public OppslagstjenesteProxy(OppslagstjenesteKonfigurasjon konfigurasjon, OppslagstjenesteInstillinger oppslagstjenesteInstillinger)
+        {
+            _oppslagstjenesteInstillinger = oppslagstjenesteInstillinger;
+            OppslagstjenesteKonfigurasjon = konfigurasjon;
+        }
+
+        public OppslagstjenesteKonfigurasjon OppslagstjenesteKonfigurasjon { get; }
+
+        private static string NetVersion
+        {
+            get
+            {
+                var netVersion = Assembly
+                    .GetExecutingAssembly()
+                    .GetReferencedAssemblies().First(x => x.Name == "System.Core").Version.ToString();
+                return netVersion;
+            }
+        }
+
+        private static Version AssemblyVersion
+        {
+            get
+            {
+                var assemblyVersion = Assembly.GetExecutingAssembly().GetName().Version;
+                return assemblyVersion;
+            }
+        }
+
+        public async Task<T> SendAsync<T>(AbstractEnvelope envelope)
+        {
+            var httpClientHandler = new HttpClientHandler();
+            if (OppslagstjenesteKonfigurasjon.BrukProxy)
+            {
+                httpClientHandler = new HttpClientHandler
+                {
+                    Proxy = Proxy()
+                };
+            }
+
+            using (var client = new HttpClient(httpClientHandler))
+            {
+                SetDefaultRequestHeaders(client);
+                ValiderForespørsel(envelope);
+                try
+                {
+                    var requestXml = envelope.ToXml();
+                    Logger.Log(TraceEventType.Verbose, requestXml.OuterXml);
+                    var stringContent = new StringContent(requestXml.InnerXml, Encoding.UTF8, "application/soap+xml");
+                    using (var response = await client.PostAsync(OppslagstjenesteKonfigurasjon.Miljø.Url, stringContent))
+                    {
+                        var soapResponse = await response.Content.ReadAsStreamAsync();
+                        var responseDokument = new ResponseDokument(soapResponse);
+                        ValiderRespons(envelope, responseDokument);
+                        Logger.Log(TraceEventType.Verbose, responseDokument.Envelope.InnerXml);
+                        return responseDokument.TilDtoObjekt<T>();
+                    }
+                }
+                catch (WebException we)

```

Comment:
Du tar drepen på catching av exceptions om vi får SOAP-error tilbake her, for det er ikke `WebException` som kastes nå. Vi bør lage tester som sjekker at dette skjer.